src/AbstractSQLRules2SQL.ts

```diff
@@ -394,26 +394,32 @@ const mathOperatorNodeTypes = new Set([
 	'SubtractDateNumber',
 ]);
 
-const mathOpValue = (
+const precedenceSafeOpValue = (
+	parentNodeType: string,
 	valueMatchFn: MetaMatchFn,
 	args: AbstractSqlType[],
 	index: number,
 	indent: string,
 ) => {
 	const operandAbstractSql = getAbstractSqlQuery(args, index);
-	const numericValue = valueMatchFn(operandAbstractSql, indent);
+	const valueExpr = valueMatchFn(operandAbstractSql, indent);
 	const [childNodeType] = operandAbstractSql;
-	if (mathOperatorNodeTypes.has(childNodeType)) {
-		return `(${numericValue})`;
+	if (
+		(mathOperatorNodeTypes.has(parentNodeType) &&
+			mathOperatorNodeTypes.has(childNodeType)) ||
+		// We need parenthesis for chained boolean comparisons, otherwise PostgreSQL complains.
+		(parentNodeType in comparisons && childNodeType in comparisons)
```

Comment:
A case that this doesn't emit properly is `SELECT (1 = 0) BETWEEN (1 < 0) AND ((1 > 0))`, in which case it doesn't include parenthesis, which makes the expression differ from the AST.
For that we would need to define the precedence order like in:
See: https://github.com/balena-io-modules/abstract-sql-compiler/compare/184-operator-precedence-light?expand=1#diff-a0b17b1f96003c01184ccb6d784d37e5e78801c6a7ef679c1ff601df04a5d288R389